src/main/java/org/cryptomator/cryptofs/fh/OpenCryptoFile.java

```diff
@@ -183,17 +183,13 @@ public void updateCurrentFilePath(Path newFilePath) {
 		currentFilePath.updateAndGet(p -> p == null ? null : newFilePath);
 	}
 
-	private synchronized void channelClosed(CleartextFileChannel cleartextFileChannel) throws IOException {
-		try {
-			FileChannel ciphertextFileChannel = openChannels.remove(cleartextFileChannel);
-			if (ciphertextFileChannel != null) {
-				chunkIO.unregisterChannel(ciphertextFileChannel);
-				ciphertextFileChannel.close();
-			}
-		} finally {
-			if (openChannels.isEmpty()) {
-				close();
-			}
+	private synchronized void channelClosed(CleartextFileChannel cleartextFileChannel) {
+		FileChannel ciphertextFileChannel = openChannels.remove(cleartextFileChannel);
+		if (ciphertextFileChannel != null) {
+			chunkIO.unregisterChannel(ciphertextFileChannel);
+		}
+		if (openChannels.isEmpty()) {
+			close();

```

Comment:
Since closing the ciphertext file channel moved to the CleartextFileChannel, we should change the type of `openChannels` to a Set (use `ConcurrentHashMap.newKeySet()`)

Reply:
We cannot change the type of `openChannels` at the moment, because we still need the ciphertextFileChannel for removing it from the chunkIO class.

If we inject the chunkIO object to the cleartextFileChannel, we could refactor it. I would also favor this, but it is out of scope of this PR.